test/wallet-http.js

```diff
@@ -0,0 +1,650 @@
+    const json = await wallet.createOpen({
+      name: name,
+      broadcast: false
+    });
+
+    let entered = false;
+    node.mempool.on('tx', () => entered = true);
+
+    await sleep(100);
+
+    // tx is not in the mempool
+    assert.equal(entered, false);
+
+    const mtx = MTX.fromJSON(json);
+    assert.ok(mtx.hasWitness());
+
+    const tx = mtx.toTX();
+
+    // tx is valid
+    assert.ok(tx.verify(mtx.view));
+
+    const opens = tx.outputs.filter(output => output.covenant.type === types.OPEN);
+    assert.equal(opens.length, 1);
+  });
+
+  it('should create an open and not sign the transaction', async () => {
+    const json = await wallet.createOpen({
+      name: name,
+      broadcast: false,
+      sign: false
+    });
+
+    let entered = false;
+    node.mempool.on('tx', () => entered = true);
+
+    await sleep(100);
+
+    assert.equal(entered, false);
+
+    const mtx = MTX.fromJSON(json);
+
+    assert.equal(mtx.verify(), false);
+  });
+
+  it('should throw error with incompatible broadcast and sign options', async () => {
+    const fn = async () => await (wallet.createOpen({
+      name: name,
+      broadcast: true,
+      sign: false
+    }));
+
+    assert.rejects(fn, 'Must sign when broadcasting');
+  });
+
+  it('should fail to create open for account with no monies', async () => {
+    const info = await wallet.getAccount(accountTwo);
+    assert.equal(info.balance.tx, 0);
+    assert.equal(info.balance.coin, 0);
+
+    const fn = async () => (await wallet.createOpen({
+      name: name,
+      account: accountTwo
+    }));
+
+    assert.rejects(fn);
+  });
+
+  it('should mine to the account with no monies', async () => {
+    const height = 5;
+
+    const {receiveAddress} = await wallet.getAccount(accountTwo);
+
+    for (let i = 0; i < height; i++)
+      await nclient.execute('generatetoaddress', [1, receiveAddress]);
+
+    const info = await wallet.getAccount(accountTwo);
+    assert.ok(info.balance.tx, height);
+    assert.ok(info.balance.coin, height);
+  });
+
+  it('should create open for specific account', async () => {
+    const json = await wallet.createOpen({
+      name: name,
+      account: accountTwo
+    });
+
+    const info = await wallet.getAccount(accountTwo);
+
+    // assert that each of the inputs belongs to the account
+    for (const {address} of json.inputs) {
+      const keyInfo = await wallet.getKey(address);
+      assert.equal(keyInfo.name, info.name);
+    }
+  });
+
+  it('should open a bid', async () => {
+    await wallet.createOpen({
+      name: name
+    });
+
+    // save chain height for later comparison
+    const info = await nclient.getInfo();
+
+    const {treeInterval} = network.names;
+    for (let i = 0; i < treeInterval + 1; i++)
+      await nclient.execute('generatetoaddress', [1, cbAddress]);
+
+    await sleep(100);
+
+    const json = await wallet.createBid({
+      name: name,
+      bid: 1000,
+      lockup: 2000,
+      debug: true
```

Comment:
What's this `debug` doing?

Reply:
Good catch haha, was using that on the server side to pause the execution and introspect the state of things